File: tmp-mei/fgenjs5030319945734616766/base/Assert_0.js
import { base$$Sealed_0 } from "../base/Sealed_0.js";
import * as rt from "../rt/main.js";

export class base$$Assert_0 extends base$$Sealed_0 {
  static $self = new base$$Assert_0();

  $exclamation$imm(assertion_m$) {
  return base$$Assert_0.$exclamation$imm$fun(assertion_m$, this);
}


$exclamation$imm(assertion_m$, cont_m$) {
  return base$$Assert_0.$exclamation$imm$fun(assertion_m$, cont_m$, this);
}


$exclamation$imm(assertion_m$, msg_m$, cont_m$) {
  return base$$Assert_0.$exclamation$imm$fun(assertion_m$, msg_m$, cont_m$, this);
}


_fail$imm() {
  return base$$Assert_0._fail$imm$fun(this);
}


_fail$imm(msg_m$) {
  return base$$Assert_0._fail$imm$fun(msg_m$, this);
}

  static $exclamation$imm$fun(assertion_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static $exclamation$imm$fun(assertion_m$, cont_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static $exclamation$imm$fun(assertion_m$, msg_m$, cont_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static _fail$imm$fun($this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

    static _fail$imm$fun(msg_m$, $this) {
  return (function() {
  console.error("Program aborted at:\n" + new Error().stack);
  if (typeof process !== "undefined") process.exit(1);
  else throw new Error("Program aborted");
})()
;
}

}

File: tmp-mei/fgen4255027576711118260/base/Assert_0.java
package base;
public interface Assert_0 extends base.Sealed_0{
Assert_0 $self = new Assert_0Impl();
base.Void_0 $exclamation$imm(base.Bool_0 assertion_m$);

Object $exclamation$imm(base.Bool_0 assertion_m$, base.AssertCont_1 cont_m$);

Object $exclamation$imm(base.Bool_0 assertion_m$, rt.Str msg_m$, base.AssertCont_1 cont_m$);

Object _fail$imm();

Object _fail$imm(rt.Str msg_m$);
static base.Void_0 $exclamation$imm$fun(base.Bool_0 assertion_m$, base.Assert_0 $this) {
  return ((base.Void_0)(switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (base.Void_0) null;
}})
);
}

static Object $exclamation$imm$fun(base.Bool_0 assertion_m$, base.AssertCont_1 cont_m$, base.Assert_0 $this) {
  return (switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (Object) null;
}})
;
}

static Object $exclamation$imm$fun(base.Bool_0 assertion_m$, rt.Str msg_m$, base.AssertCont_1 cont_m$, base.Assert_0 $this) {
  return (switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (Object) null;
}})
;
}

static Object _fail$imm$fun(base.Assert_0 $this) {
  return (switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (Object) null;
}})
;
}

static Object _fail$imm$fun(rt.Str msg_m$, base.Assert_0 $this) {
  return (switch (1) { default -> {
  System.err.println("Program aborted at:\n"+java.util.Arrays.stream(Thread.currentThread().getStackTrace()).map(StackTraceElement::toString).collect(java.util.stream.Collectors.joining("\n")));
  System.exit(1);
  yield (Object) null;
}})
;
}
}